### PR TITLE
Move l3bitset to l3kernel (stable)

### PR DESCRIPTION
**Pull request by josephwright**

### CODE REVIEW

File: l3kernel/doc/source3body.tex [hunk 1]
```diff
@@ -588,6 +588,7 @@ \part{Data types}
     l3fp-functions.dtx
   }
 \ExplSyntaxOff
+\DocInput{l3bitset.dtx}
```
Comment (muzimuzhi) [inline l3kernel/doc/source3body.tex:591]: Any better position than just between `fp` submodules and `fparray`?

File: l3kernel/l3bitset.dtx [hunk 2]
```diff
@@ -119,40 +121,39 @@
 %         \meta{name2} |=| \meta{index2} |,| \ldots{}
 %      \}
```
Comment (muzimuzhi) [inline l3kernel/l3bitset.dtx:122]: From `texdoc l3bitset`, indents are missing in this and several more `syntax` envs.

**Review by muzimuzhi — COMMENTED** [review]
Not a complete review.

New `bitset` variable type needs to be added to the full type list in `source3body.tex` (sec. 1.1 "Naming functions and variables" in `interface3.pdf`).

File: l3kernel/l3bitset.dtx (hunk 2 above)
Comment (josephwright) [reply l3kernel/l3bitset.dtx:122]: That we can deal with post-PR I think

File: l3kernel/doc/source3body.tex (hunk 1 above)
Comment (josephwright) [reply l3kernel/doc/source3body.tex]: Moved

**Comment by josephwright** [comment]
> Not a complete review.
> 
> New `bitset` variable type needs to be added to the full type list in `source3body.tex` (sec. 1.1 "Naming functions and variables" in `interface3.pdf`).

Added

**Comment by car222222** [comment]
Not quite what I suggested, but it does remove the error.

**Comment by josephwright** [comment]
> Not quite what I suggested, but it does remove the error.

We already had string so I thought a bit of variety was good ...

**Comment by car222222** [comment]
Sure, but my version any need for another name.
It also adds an extra detail at the end.

**Comment by u-fischer** [comment]
l3bitset is as far as I can see only used in the pdfmanagement, so I can perpare for the move and add a test to load it only if not already in the kernel. But generally: should we add some empty wrapper for a transition time if we move experimental modules?

**Comment by car222222** [comment]
Thanks:-)

**Comment by car222222** [comment]
What @u-fischer wrote makes perfect sense to me.
There may be other ways to achieve this?

**Comment by muzimuzhi** [comment]
> If you upload l3kernel with this PR to ctan I will get an error in the pdfmanagement as it uses \RequirePackage{l3bitset}. This could be avoided if there would be (temporary) a small dummy l3bitset.sty which simply issues a warning "do not used this package anymore, code has been moved into l3kernel".

Other packages to be moved from `l3experimental` to `l3kernel` would need similar treatments.

**Comment by josephwright** [comment]
@u-fischer  Oh, that. In the past we did have a stub setup, but that was when we were consolidating widely-used modules. Here, I was imagining you'd go with
```latex
\cs_if_exist:NF \bitset_new:Nn
  { \RequirePackage { l3bitset } }
```
or similar.

**Comment by u-fischer** [comment]
> wouldn't it be better to have `\ProvidesPackage` somewhere?

That wouldn't help. I mean during the "test-PR-phase" the package is still there (in the texmf-dist tree). 

>  Either way, how long would you want this to 'hang around'?

I would remove the package in the next update of the pdf management whenever that it is (currently there is nothing pending). And then l3kernel could remove a stub whenever you like.

**Comment by josephwright** [comment]
@u-fischer  Latest change work for you? If so, it's likely a model for other moves from `l3experimental` if/when we make them (which I would of course like ...)

**Comment by josephwright** [comment]
@FrankMittelbach  I'm trying to avoid having to have sources to DocStrip, etc., which 'come and go' ..

**Comment by FrankMittelbach** [comment]
not the worst argument :-)

**Pull request merged**